Sources/Plasma/PubUtilLib/plMath/plTriUtils.cpp

```diff
@@ -45,17 +45,18 @@ You can contact Cyan Worlds, Inc. by email legal@cyan.com
 #include <cmath>
 #include "plTriUtils.h"
 
-static const float kAlmostZero = 1.e-5f;
-static const float kPastZero = -kAlmostZero;
-static const float kPastOne = 1.f + kAlmostZero;
-static const float kAlmostOne = 1.f - kAlmostZero;
-static const float kAlmostZeroSquared = kAlmostZero*kAlmostZero;
+static constexpr float kAlmostZero = 1.e-5f;
+static constexpr float kPastZero = -kAlmostZero;
+static constexpr float kPastOne = 1.f + kAlmostZero;
+static constexpr float kAlmostOne = 1.f - kAlmostZero;
+static constexpr float kAlmostZeroSquared = kAlmostZero*kAlmostZero;
 
 static inline hsVector3 Cross(const hsScalarTriple& p0, const hsScalarTriple& p1)
 {
-    return hsVector3(p0.fY * p1.fZ - p0.fZ * p1.fY, 
-                    p0.fZ * p1.fX - p0.fX * p1.fZ, 
-                    p0.fX * p1.fY - p0.fY * p1.fX);
+    return {p0.fY * p1.fZ - p0.fZ * p1.fY, 
+    	       p0.fZ * p1.fX - p0.fX * p1.fZ, 
+    	       p0.fX * p1.fY - p0.fY * p1.fX
+    };
```

Comment:
This should probably remain an `hsVector3` for type safety and clarity. I don't think there's any performance impact here.

Reply:
fair enough i just found it as duplicating the type since the function already returns it.

Reply:
Always better to use the defined constructor for an object instead of pushing in memory directly. Even if it's a bit duplicative with the type name.

Reply:
@dgelessus - I see your emoji. I'm poking at this and I see C++ is policing the number of elements in the struct.  However - this struct does have a defined constructor and it does seem weird to go around it.

Reply:
Ah, I wanted to explain later and forgot, sorry. As I understand it, the brace syntax in this case *will* use the constructor defined in the class, despite the brace syntax. According to cppreference, the C-style field by field initialization is called [aggregate initialization](https://en.cppreference.com/w/cpp/language/aggregate_initialization), which never applies to classes with user-defined constructors. Instead, the brace syntax does [list initialization](https://en.cppreference.com/w/cpp/language/list_initialization) here, which in this case calls the normal constructor, exactly like the previous code.

Reply:
@dgelessus Thanks! I'll review the docs.